src/db.c

```diff
@@ -2265,15 +2265,29 @@ int migrateGetKeys(struct redisCommand *cmd, robj **argv, int argc, getKeysResul
     num = 1;
 
     /* But check for the extended one with the KEYS option. */
-    if (argc > 6) {
+    struct {
+        char* name;
+        int skip;
+    } skiplist[] = {       
+        {"copy", 0},
+        {"replace", 0},
+        {"auth", 1},
+        {"auth2", 2},
+        {NULL, 0}
+    };
+    if (argc > 6 && sdslen(argv[3]->ptr) == 0) {
```

Comment:
move the key argument check into the "keys" options check scope, and return 0 if the key argument is not empty, then migrate can return the syntax error when executing.

Reply:
In function `migrateCommand` we already have this. Should we move it in advance?
```
else if (!strcasecmp(c->argv[j]->ptr,"keys")) {
    if (sdslen(c->argv[3]->ptr) != 0) {
        addReplyError(c,
            "When using MIGRATE KEYS option, the key argument"
            " must be set to the empty string");
        return;
    }
    // ...
}
```

Reply:
Oh I got it. We deliberately miss the ACL check and let `migrateCommand` return syntax error.